collector/accelerators.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type cardData struct {
+	vendor string
+	model  string
+}
+
+type vendorData struct {
+	vendorName string
+	devicesIDs map[string]string
+}
+
+type acceleratorsCollector struct {
+	pciDevicesPath string
+	logger         log.Logger
+}
+
+func init() {
+	registerCollector("accelerator", defaultEnabled, NewAcceleratorCollector)
+}
+
+// NewAcceleratorCollector returns a new Collector exposing accelerator cards count.
+func NewAcceleratorCollector(logger log.Logger) (Collector, error) {
+	return &acceleratorsCollector{
+		pciDevicesPath: filepath.Join(*sysPath, "bus/pci/devices"),
+		logger:         logger,
+	}, nil
+}
+
+var (
+	acceleratorCardsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "accelerator", "cards"),
+		"Number of accelerator cards per type",
+		[]string{"vendor", "model"}, nil,
+	)
+
+	nvidiaDeviceIDsMap = map[string]string{
```

Comment:
Note for a follow-up but I can imagine a configuration file (json or yaml) managed by CMO which would contain the mapping. This way enriching the database would only require a change in CMO.